app/js/actions/drug.js

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import * as actionTypes from './actionTypes';
+
+const contextPath = window.location.href.split('/')[3];
+const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
+
+export const searchDrugSuccess = drugs => ({
+  type: actionTypes.SEARCH_DRUGS_SUCCESS,
+  drugs,
+});
+
+export const searchDrugFailure = error => ({
+  type: actionTypes.SEARCH_DRUGS_FAILURE,
+  error,
+});
+
+export const searchDrugLoading = status => ({
+  type: actionTypes.SEARCH_DRUGS_LOADING,
+  status,
+});
+
+export const searchDrug = text => (dispatch) => {
+  dispatch(searchDrugLoading(true));
+  return axios.get(`${apiBaseUrl}/drug?q=${text}`)
+    .then((response) => {
+      dispatch(searchDrugLoading(false));
+      dispatch(searchDrugSuccess(response.data.results));
+    })
+    .catch((error) => {
+      dispatch(searchDrugLoading(false));
+      if (error.response) {
+        dispatch(searchDrugFailure(error.response));
+      } else {
+        dispatch(searchDrugFailure({ data: { message: "Network connection error" } }));
```

Comment:
How do you tell that this is a network connection error?

Reply:
Based on some research on axios, I figured that when there is an error with an axios call and there is no corresponding error.response, it means that the request was not successfully made to the server -  a situation caused by network connection failure